mock-plugin/src/main/java/io/mosip/signup/plugin/mock/service/MockProfileRegistryPluginImpl.java

```diff
@@ -5,51 +5,279 @@
+		}
+	}
+    
+    private MockIdentityRequest buildIdentityRequest(JsonNode inputJson){
+    	MockIdentityRequest mockIdentityRequest = new MockIdentityRequest();
+    	Iterator<Map.Entry<String, JsonNode>> itr = inputJson.fields();
+    	List<String> fieldNames = new ArrayList<>();
+    	Field[] fields = mockIdentityRequest.getClass().getDeclaredFields();
+    	for(Field field:fields) {
+    		fieldNames.add(field.getName());
+    	}
+    	while(itr.hasNext()) {
+    		String field = ((TextNode)itr.next()).textValue();
+    		if(fieldNames.contains(field)) {
+    			try {
+	    			if(field.equalsIgnoreCase(PASSWORD)) {
+	    				Password password = generateSaltedHash(inputJson.get("password").asText());
+						BeanUtils.setProperty(mockIdentityRequest, field, password);
+	
+	    			}
+	    			else if(langBasedFields.contains(field)) {
+	    				List<LanguageValue> fieldValue = objectMapper.readValue(inputJson.get(field).toString(), 
+									new TypeReference<List<LanguageValue>>() {});
+	    				BeanUtils.setProperty(mockIdentityRequest, field, fieldValue);
+	    			}
+	    			else {
+	    				BeanUtils.setProperty(mockIdentityRequest, field, inputJson.get(field).toString());
+	    			}
+	    		}catch(IllegalAccessException|InvocationTargetException|JsonProcessingException ex) {
+	    			ex.printStackTrace();
+	    		}
+    		}
+    	}
+    	
+    	return mockIdentityRequest;
+    }
+    
+    private UUID getUniqueIdentifier() throws ProfileException {
+    	return UUID.randomUUID();
+
+    }
+    
+    private MockIdentityResponse addIdentity(MockIdentityRequest identityRequest) throws ProfileException{
+        RequestWrapper<MockIdentityRequest> restRequest = new RequestWrapper<>();
+        restRequest.setRequestTime(getUTCDateTime());
+        restRequest.setRequest(identityRequest);
+        ResponseWrapper<MockIdentityResponse> responseWrapper = request(identityEndpoint, HttpMethod.POST, restRequest,
+                new ParameterizedTypeReference<ResponseWrapper<MockIdentityResponse>>() {});
+        return responseWrapper.getResponse();
+    }
+    
+    private MockIdentityResponse updateIdentity(MockIdentityRequest identityRequest) throws ProfileException{
+        RequestWrapper<MockIdentityRequest> restRequest = new RequestWrapper<>();
+        restRequest.setRequestTime(getUTCDateTime());
+        restRequest.setRequest(identityRequest);
+        ResponseWrapper<MockIdentityResponse> responseWrapper = request(identityEndpoint, HttpMethod.PUT, restRequest,
+                new ParameterizedTypeReference<ResponseWrapper<MockIdentityResponse>>() {});
+        return responseWrapper.getResponse();
+    }
+    
+    private String getUTCDateTime() {
+        return ZonedDateTime
+                .now(ZoneOffset.UTC)
+                .format(DateTimeFormatter.ofPattern(UTC_DATETIME_PATTERN));
+    }
+    
+    private Password generateSaltedHash(String password) throws ProfileException {
```

Comment:
When esignet is integrated with mock-identity-system, It is not required to call an endpoint to generate the pwd hash. can you check how is the pwd stored in the mock-identity-system?

Reply:
As of now, we don't have pwd login with mock-identity-system, only pin and otp were supported with it. Should I change the password encryption with a simple algorithm as it is for mock?